sql/parser/eval.go

```diff
@@ -259,6 +263,147 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// DDate is the date Datum.
+type DDate struct {
```

Comment:
I'm sure that's intentional, though I don't know what the "killer argument" is. Being able to re-define existing methods on the type "alias" (Pete doesn't like to call it that, and I tend to think he's right) is one advantage (among other things, this allows exposing only a subset of the original methods by simply passing through).